### DISCUSSION TITLE
Getting a timeout on the Vercel Hobby plan

### DISCUSSION THREAD

**Discussion by adamchipperfield** [nextauthjs/next-auth#4288] [Help]
### Question 💬

I have a Next.js app with next-auth up and running locally, it works perfectly. But when I deploy to Vercel I get a "FUNCTION_INVOCATION_TIMEOUT" error when I try to sign in (as a user via next-auth).

I did some digging and I _think_ it's because I'm on the Hobby plan on Vercel, and the wait for functions is something like 5 seconds.

Is there a way I can still use next-auth on a Hobby plan?

### How to reproduce ☕️

Just deploy a next-auth app on Vercel's Hobby plan.

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by adamchipperfield** [comment]
Turns out the same is happening on the Pro plan too. Any ideas why I'm getting a timeout once deploying to Vercel?

**Reply by ThangHuuVu** [reply]
Possibly related: https://github.com/nextauthjs/next-auth/issues/1535

**Comment by redimongo** [comment]
Seems Next-Auth.js is very slow. When I put it on my VPS server it works, but when it's on Vercel is does not work - timeout.
Clearly calling so many functions at once is causing something to slow down.

**Comment by BranonConor** [comment]
I'm also having this issue. In local development it all works snappy and quick, and I went to get it setup on Vercel to test it out in production and I can't get past the last google auth step because it times out every time. I wondered about upping to pro so the timeout is 60s vs 10s on the hobby plan, but my worry is it won't fix it. Open to ideas!

**Comment by redimongo** [comment]
Yes it seems Vercel is really only interested in "FAST" response time.
saying this if you run your app on say DigitalOcean, or another cloud
provider it works as intended.

This may not be an issue with Next-Auth it maybe a problem with Vercel.
*Russell Harrower*
*Radio Media PTY LTD*
0458 375 272

**Comment by JanderSilv** [comment]
Any updates on this issue?

**Comment by suitux** [comment]
Any update on this?

**Comment by JanderSilv** [comment]
In my case, it was the @mui/icons-material.

Solution: https://github.com/mui/material-ui/issues/35450

**Reply by ajhenry** [reply]
You finally saved me from hours of debugging with the vercel and prisma team, thanks 🙇
